packages/block-editor/src/components/rich-text/index.js

```diff
@@ -179,11 +179,17 @@ export function RichTextWrapper(
 			const blockBindingsSource = getBlockBindingsSource(
 				relatedBinding.source
 			);
+			const blockBindingsContext = {};
```

Comment:
I wanted to highlight the differences I discovered with how the code feeds other API methods with the context.

https://github.com/WordPress/gutenberg/blob/b681b5063add14746f7cee34de21493532e401b5/packages/block-editor/src/hooks/use-bindings-attributes.js#L120

https://github.com/WordPress/gutenberg/blob/b681b5063add14746f7cee34de21493532e401b5/packages/block-editor/src/hooks/use-bindings-attributes.js#L139-L142

https://github.com/WordPress/gutenberg/blob/b681b5063add14746f7cee34de21493532e401b5/packages/block-editor/src/hooks/use-bindings-attributes.js#L162-L167

https://github.com/WordPress/gutenberg/blob/b681b5063add14746f7cee34de21493532e401b5/packages/block-editor/src/hooks/use-bindings-attributes.js#L238-L248

While `updatedContext` is correct in that code as it eventually gets passed to the `<Edit />` component, I don't think we should use `updatedContext` when passing context to `setValues` and `getValues`. Instead, we should replicate the handling proposed in this PR. In effect, `useSelect` defined for the source on the server and/or client would be the ultimate contract of what these API methods can get as params.

I would proceed with this PR as planned. If we agree the same changes should get applied in other places, then there should be two subsets of the context:
- `blockBindingsContext ` like in this PR that gets passed to the methods registered for the source
- `updatedContext` would be refactored to: `const updatedContext = { ...context,  ... blockBindingsContext };

Reply:
Unifying and using only the context defined in `usesContext` for `setValues` and `getValues` makes sense to me. I can work on that in a follow-up.

Reply:
I'm exploring changing this in [this pull request](https://github.com/WordPress/gutenberg/pull/65661). I believe just changing how `updatedContext` is initialized should be enough.